pkg/generate/app/cmd/newapp.go

```diff
@@ -334,6 +334,14 @@ func (c *AppConfig) addReferenceBuilderComponents(b *app.ReferenceBuilder) {
 		input.Searcher = searcher
 		return input
 	})
+
+	_, repos, _ := b.Result()
+	for _, repo := range repos {
+		repo.SetContextDir(c.ContextDir)
+		if c.Strategy == "docker" {
```

Comment:
This seems like it should be in infer build types - if we need to pass repositories to that method we should